### PR TITLE
Fix Seq strip method regression

### PR DESCRIPTION
**Pull request by peterjc**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Address in part #3313.

### CODE REVIEW

**Review by mdehoon — APPROVED** [review]

File: Bio/Seq.py [hunk 1]
```diff
@@ -395,7 +398,11 @@ def count_overlap(self, sub, start=0, end=sys.maxsize):
         HOWEVER, do not use this method for such cases because the
         count() method is much for efficient.
         """
-        sub_str = str(sub)
+        if isinstance(sub, (Seq, MutableSeq)):
+            sub_str = str(sub)
```
Comment (mdehoon) [inline Bio/Seq.py:402]: or use `sub = str(sub)`, then we can drop the `else` statement.

Comment (peterjc) [reply Bio/Seq.py:402]: Yes, if we changed the while loop to use ``sub`` that would work. I thought this was clearer retaining the cast variables ``self_str`` and ``sub_str``.

Comment (mdehoon) [reply Bio/Seq.py:402]: Up to you then. I am fine either way.

Comment (peterjc) [reply Bio/Seq.py]: I'll change it. They are both clear, but your proposal is shorter.

**Comment by mdehoon** [comment]
Can this be merged?

**Comment by peterjc** [comment]
Done, didn't immediately to give you a chance to comment on the final changes. Thank you.

File: Bio/Seq.py [hunk 2]
```diff
@@ -610,11 +644,34 @@ def strip(self, chars=None):
...
+           ...
+        TypeError: strip arg must be None or str
 
         See also the lstrip and rstrip methods.
         """
-        return Seq(str(self).strip(str(chars)))
+        if isinstance(chars, (Seq, MutableSeq)):
+            return Seq(str(self).strip(str(chars)))
+        else:
+            # Want TypeError on None, int, etc
```
Comment (mdehoon) [inline Bio/Seq.py:673]: Not on `None`

File: Bio/Seq.py [hunk 3]
```diff
@@ -625,11 +682,16 @@ def lstrip(self, chars=None):
...
+        >>> Seq("AAACGTA").lstrip("A")
+        Seq('CGTA')
 
         See also the strip and rstrip methods.
         """
-        return Seq(str(self).lstrip(str(chars)))
+        if isinstance(chars, (Seq, MutableSeq)):
+            return Seq(str(self).lstrip(str(chars)))
+        else:
+            # Want TypeError on None, int, etc
```
Comment (mdehoon) [inline Bio/Seq.py:693]: Same here

File: Bio/Seq.py [hunk 4]
```diff
@@ -651,7 +713,11 @@ def rstrip(self, chars=None):
 
         See also the strip and lstrip methods.
         """
-        return Seq(str(self).rstrip(str(chars)))
+        if isinstance(chars, (Seq, MutableSeq)):
+            return Seq(str(self).rstrip(str(chars)))
+        else:
+            # Want TypeError on None, int, etc
```
Comment (mdehoon) [inline Bio/Seq.py:719]: same

**Pull request merged**